src/core/api/media_element_track_choice_manager.ts

```diff
@@ -0,0 +1,471 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BehaviorSubject } from "rxjs";
+import EventEmitter from "../../utils/event_emitter";
+import normalizeLanguage from "../../utils/languages";
+import {
+  IAudioTrackPreference,
+  ITextTrackPreference,
+  ITMAudioTrack,
+  ITMAudioTrackListItem,
+  ITMTextTrack,
+  ITMTextTrackListItem,
+  ITMVideoTrack,
+  ITMVideoTrackListItem,
+} from "./track_choice_manager";
+
+// TODO w3c defines an onremovetrack attribute which is not present on
+// ts type definition
+export interface ICustomTextTrackList extends TextTrackList {
```

Comment:
That's an `ICompat`, not an `ICustom`.
`ICompat` is for a difference between what TS says and what some browsers do.
`ICustom` is for a custom implementation of a native thing in our code.

Also should be put in `browser_compatibility_types.ts`

Reply:
done